### PR TITLE
fix: Error messages must contain all the error, including the initial error code (ERR)

### LINKED ISSUE #439: On RedisModuleReplyWithError error messages must contain ERR
Error messages must contain all the error, including the initial error code (ERR), as stated on the RedisModule_ReplyWithError documentation:

````
/* Reply with the error 'err'.
 *
 * Note that 'err' must contain all the error, including
 * the initial error code. The function only provides the initial "-", so
 * the usage is, for example:
 *
 *     RedisModule_ReplyWithError(ctx,"ERR Wrong Type");
 *
 * and not just:
 *
 *     RedisModule_ReplyWithError(ctx,"Wrong Type");
 *
 * The function always returns REDISMODULE_OK.
 */
int RM_ReplyWithError(RedisModuleCtx *ctx, const char *err) {
    return replyWithStatus(ctx,err,"-");
}
````

## Importance to the clients

This is important given that is breaking clients parsing RESP and properly handling it. 
An example of it radix client, which is expecting the ERR to be contained on resp reply and given that it is not is causing errors like this one:
```
panic: should never get here

goroutine 1 [running]:
github.com/mediocregopher/radix/v3/resp/resp2.saneDefault(...)
        /home/filipe/go/pkg/mod/github.com/mediocregopher/radix/v3@v3.5.1/resp/resp2/resp.go:789
github.com/mediocregopher/radix/v3/resp/resp2.Any.UnmarshalRESP(0x610f00, 0xc000098720, 0xc0000b0000, 0xc0000b85a0, 0x650b60, 0xc0000dbcc0)
        /home/filipe/go/pkg/mod/github.com/mediocregopher/radix/v3@v3.5.1/resp/resp2/resp.go:828 +0xcb1
github.com/mediocregopher/radix/v3.(*cmdAction).UnmarshalRESP(0xc0000d4150, 0xc0000b85a0, 0x0, 0x0)
        /home/filipe/go/pkg/mod/github.com/mediocregopher/radix/v3@v3.5.1/action.go:269 +0x4c
github.com/mediocregopher/radix/v3.(*connWrap).Decode(0xc0000a62e0, 0x6b55a0, 0xc0000d4150, 0x0, 0x0)
```
that is originated from:
```

func saneDefault(prefix byte) interface{} {
	// we don't handle ErrorPrefix because that always returns an error and
	// doesn't touch I
	switch prefix {
	case ArrayPrefix[0]:
		ii := make([]interface{}, 8)
		return &ii
	case BulkStringPrefix[0]:
		bb := make([]byte, 16)
		return &bb
	case SimpleStringPrefix[0]:
		return new(string)
	case IntPrefix[0]:
		return new(int64)
	}
	panic("should never get here")
}
```

### PR DESCRIPTION
**Pull request by filipecosta90**
fixes #439 . This is important for clients properly parsing the RESP protocol

### CODE REVIEW

**Comment by ashtul** [comment]
> fixes #439 . This is important for clients properly parsing the RESP protocol

Did anything change? We are not doing it in other modules.

**Comment by filipecosta90** [comment]
> Did anything change? We are not doing it in other modules.

We're doing it in AI at least and should move towards ensuring this behavior on all modules, given that this is the right approach as stated on the docs:
```
/* Reply with the error 'err'.
 *
 * Note that 'err' must contain all the error, including
 * the initial error code. The function only provides the initial "-", so
 * the usage is, for example:
 *
 *     RedisModule_ReplyWithError(ctx,"ERR Wrong Type");
 *
 * and not just:
 *
 *     RedisModule_ReplyWithError(ctx,"Wrong Type");
 *
 * The function always returns REDISMODULE_OK.
 */
int RM_ReplyWithError(RedisModuleCtx *ctx, const char *err) {
    return replyWithStatus(ctx,err,"-");
}
```

File: src/common.h [hunk 1]
```diff
@@ -1,2 +1,6 @@
 
 #pragma once
+
+#define ERR "ERR"
```
Comment (rafie) [inline src/common.h:4]: `ERR` is a little broad-spectrum when defined in a header. Maybe `TS_ERR`?

Comment (danni-m) [reply src/common.h]: fixed

File: src/module.c [hunk 2]
```diff
@@ -699,19 +700,19 @@ static int internalAdd(RedisModuleCtx *ctx,
     timestamp_t lastTS = series->lastTimestamp;
     uint64_t retention = series->retentionTime;
     // ensure inside retention period.
-    if (retention && timestamp < lastTS && retention < lastTS - timestamp) {
-        RedisModule_ReplyWithError(ctx, "TSDB: Timestamp is older than retention");
+    if (retention && timestamp < lastTS && timestamp < lastTS - retention) {
+        RTS_ReplyGeneralError(ctx, "TSDB: Timestamp is older than retention");
         return REDISMODULE_ERR;
```
Comment (ashtul) [inline src/module.c:705]: why?

**Pull request merged**